lib/manager/components/DeploymentSettings.js

```diff
@@ -198,22 +214,55 @@ class DeploymentSettings extends Component<Props, State> {
     .every(key => shallowEqual(this.state[key], this.props.project[key]))
 
   render () {
-    const updaters = objectPath.get(this.state, 'routerConfig.updaters') || []
+    const updaters = get(this.state, 'routerConfig.updaters') || []
     const {project, editDisabled} = this.props
+    const {pinnedDeploymentId} = project
+    const pinnedDeployment = pinnedDeploymentId && project.deployments && project.deployments.find(d => d.id === pinnedDeploymentId)
```

Comment:
I am a bit confused by the workflow for setting auto-deploy it involves switching between two different tabs. Should the auto-deploy toggle be shown next to the pin button on the Deployments tab instead?

Reply:
No, but perhaps the pinned deployment selector should just be moved here. The tricky design issue is... pinned deployments impact not only auto-deploy, but also the current "deployed status" for the list of feed sources in the project viewer. I'm open to ideas about how to intuitively treat all of this... but ultimately, this is a "set it and forget it" kind of setting, so if the user has to switch between tabs once, it's not really that big of a deal.

Reply:
I agree with Binh that the workflow is very confusing. However, I think we ought to rethink deployments in a larger refactor where we can do other things like keep a history of feed versions that were included in each deployment. We could address that issue during that time.

Reply:
I'm not sure there's much overlap between this setting and the history of feed versions (which would more so be colocated with the DeploymentViewer). I'll think through some options for possibly moving all deployment settings to the deployments list page.